### PR TITLE
Upload source files to generate a CPG.

### PR DESCRIPTION
**Pull request by GlassAndOneHalf**
Files are limited to 10MB in size each. There is no currently no limit on the maximum number of files.

Uploaded files are streamed  to disk so running out of heap should not be a major concern 🤞.

Other changes include:
- Update Swagger to document the new route.
- Add a fancy new banner on startup.

### CODE REVIEW

File: cpgserver/src/test/scala/io/shiftleft/cpgserver/route/CpgRouteSpec.scala [hunk 1]
```diff
@@ -109,6 +111,66 @@ class CpgRouteSpec extends Http4sSpec {
...
+        .withEntity(requestContent)
+      val response = route.run(request)
+
+      check(response,
+            Status.BadRequest,
+            Some(ApiError("At least one 'file' must be specified for the CPG to be created"))) shouldBe true
+     }
+
+    "fail with 400 BAD REQUEST if a 'file' is missing a 'filename'" in withRoute() { _ =>
+      ???
```
Comment (GlassAndOneHalf) [inline cpgserver/src/test/scala/io/shiftleft/cpgserver/route/CpgRouteSpec.scala]: Just need to implement this, then we are good to go!

File: cpgserver/src/main/scala/io/shiftleft/cpgserver/route/CpgRoute.scala [hunk 2]
```diff
@@ -41,6 +49,58 @@ final class CpgRoute(cpgProvider: CpgProvider, cpgQueryExecutor: ServerAmmoniteE
...
+  private def fileSizeCheck(partSizeLimit: Long): Pipe[IO, Byte, Byte] = in => {
+    in.zipWithIndex.flatMap {
+      case (byte, count) =>
+        if (count + 1 >= partSizeLimit) {
+          fs2.Stream.raiseError[IO](FileSizeException(s"A provided file is larger than [$partSizeLimit] bytes."))
+        } else fs2.Stream.emit(byte)
+    }
+  }
+
+  import java.nio.file.{Path => JPath}
```
Comment (hubertp) [inline cpgserver/src/main/scala/io/shiftleft/cpgserver/route/CpgRoute.scala]: should be in the header

File: cpgserver/src/main/scala/io/shiftleft/cpgserver/route/CpgRoute.scala [hunk 3]
```diff
@@ -103,29 +173,36 @@ final class CpgRoute(cpgProvider: CpgProvider, cpgQueryExecutor: ServerAmmoniteE
...
+
   object CpgHttpErrorHandler extends HttpErrorHandler {
     private val logger = LoggerFactory.getLogger(this.getClass)
 
     override def handle(routes: HttpRoutes[IO]): HttpRoutes[IO] =
       HttpErrorHandler(routes) {
         case _: MessageBodyFailure =>
           BadRequest(ApiError("Invalid payload. Please check that the payload is formatted correctly.").asJson)
-        case ex =>
+        case NonFatal(ex) =>
```
Comment (hubertp) [inline cpgserver/src/main/scala/io/shiftleft/cpgserver/route/CpgRoute.scala]: non-exhaustive check doesn't complain?

Comment (GlassAndOneHalf) [reply cpgserver/src/main/scala/io/shiftleft/cpgserver/route/CpgRoute.scala]: This takes a `PartialFunction` :)

Comment (GlassAndOneHalf) [reply cpgserver/src/main/scala/io/shiftleft/cpgserver/route/CpgRoute.scala]: Aha, so I'm wrong! Nice spot, will fix it.

**Pull request merged**